src/wallet/scriptpubkeyman.cpp

```diff
@@ -820,6 +820,31 @@ bool LegacyScriptPubKeyMan::GetPubKey(const CKeyID &address, CPubKey& vchPubKeyO
     return GetWatchPubKey(address, vchPubKeyOut);
 }
 
+GCSFilter::ElementSet LegacyScriptPubKeyMan::GetAllRelevantScriptPubKeys() const
+{
+    LOCK(cs_KeyStore);
+    GCSFilter::ElementSet ret;
+    for (const auto& k : mapKeys) {
```

Comment:
What about `mapCryptedKeys`? I don't think `mapKeys` is used on encrypted wallets?

Reply:
> What about `mapCryptedKeys`? I don't think `mapKeys` is used on encrypted wallets?

Yes, this should just loop over `mapCryptedKeys` as well. `mapKeys` should be empty in encryped wallets.

Reply:
This function in general does not match everything that `IsMine` would match.In particular it doesn't enumerate all possible combinations of multisig policies with all combinations of pubkeys or all of the weird nested scripts that are possible. There probably needs to be a release note that if you have some weird and non-standard script stuff, the fast rescan won't see those things.

Reply:
Such a warning makes sense; people with such wallets will presumably understand. And descriptor wallets won't have this problem.

Reply:
`IsMine` itself won't test weird combinations. I don't see why fast rescan shouldn't be able to check for everything `IsMine` does... users shouldn't be exposed to internal implementation details like this.